ibm/service/cos/resource_ibm_cos_replication_configuration.go

```diff
@@ -0,0 +1,438 @@
+
+	bucketLocation := d.Get("bucket_location").(string)
+	endpointType := d.Get("endpoint_type").(string)
+
+	bxSession, err := meta.(conns.ClientSession).BluemixSession()
+	if err != nil {
+		return err
+	}
+
+	s3Client, err := getS3ClientSession(bxSession, bucketLocation, endpointType, instanceCRN)
+
+	replicationRule := d.Get("replication_rule")
+
+	// BUCKET REPLICATION CONFIGURATION
+	var bktrep_prefix, bktrep_status, del_marker_rep_status, rule_id, dest_bt_crn string
+	var bktrep_priority int64
+	replicateList := replicationRule.([]interface{})
+	for _, l := range replicateList {
+		replicateMap, _ := l.(map[string]interface{})
+		//Rule ID
+		if rule_idSet, exist := replicateMap["rule_id"]; exist {
+			id := rule_idSet.(string)
+			rule_id = id
+		}
+
+		//Status Enable/Disable
+		if statusSet, exist := replicateMap["enable"]; exist {
+			StatusEnabled := statusSet.(bool)
+			if StatusEnabled == true {
+				bktrep_status = "Enabled"
+			} else {
+				bktrep_status = "Disabled"
+			}
+		}
+		//Days
+		if priorSet, exist := replicateMap["priority"]; exist {
+			replicate_priority := int64(priorSet.(int))
+			bktrep_priority = replicate_priority
+		}
+		//Expire Prefix
+		if PrefixClassSet, exist := replicateMap["prefix"]; exist {
+			prefix_check := PrefixClassSet.(string)
+			bktrep_prefix = prefix_check
+		}
+		//DeleteMarkerReplicationStatus
+		if delMarkerStatusSet, exist := replicateMap["deletemarker_replication_status"]; exist {
+			del_marker_status_value := delMarkerStatusSet.(bool)
+			if del_marker_status_value == true {
+				del_marker_rep_status = "Enabled"
+			} else {
+				del_marker_rep_status = "Disabled"
+			}
+		}
+		//Destination CRN
+		if dest_bucket_CrnSet, exist := replicateMap["destination_bucket_crn"]; exist {
+			dest_bucketcrn_value := dest_bucket_CrnSet.(string)
+			dest_bt_crn = dest_bucketcrn_value
+		}
+	}
+
+	putBucketReplicationInput := &s3.PutBucketReplicationInput{
+		Bucket: aws.String(bucketName),
+		ReplicationConfiguration: &s3.ReplicationConfiguration{
+			Rules: []*s3.ReplicationRule{
+				{
+					ID:       aws.String(rule_id),
+					Status:   aws.String(bktrep_status),
+					Priority: aws.Int64(bktrep_priority),
+					Filter: &s3.ReplicationRuleFilter{
+						Prefix: aws.String(bktrep_prefix),
+					},
+					Destination:             &s3.Destination{Bucket: aws.String(dest_bt_crn)},
+					DeleteMarkerReplication: &s3.DeleteMarkerReplication{Status: aws.String(del_marker_rep_status)},
+				},
+			},
+		},
+	}
+	_, err = s3Client.PutBucketReplication(putBucketReplicationInput)
+
+	if err != nil {
+		return fmt.Errorf("failed to update the replication rule on COS bucket %s, %v", bucketName, err)
+	}
+
+	//Generating a fake id which contains every information about to get the bucket via s3 api
+	bktID := fmt.Sprintf("%s:%s:%s:meta:%s:%s", strings.Replace(instanceCRN, "::", "", -1), "bucket", bucketName, bucketLocation, endpointType)
```

Comment:
Can you share what all the id is composed off

Reply:
id=crn:v1:bluemix:public:cloud-object-storage:global:a/ee747e45752d4696b2d082bcf2357559:359f1538-e46e-4747-9cfd-b8898c5fea3d:bucket:testsrc0974:meta:sjc04:public